core/src/main/java/org/radargun/stages/cache/background/LogChecker.java

```diff
@@ -143,60 +145,55 @@ public void run() {
                record.setLastUnsuccessfulCheckTimestamp(Long.MIN_VALUE);
                record.setLastSuccessfulCheckTimestamp(System.currentTimeMillis());
             } else {
+               // Perform only once in a while (every 10 unsuccessful check attempts) in order not to create much overhead
+               if (record.getUnsuccessfulCheckAttempts() > 0 && (record.getUnsuccessfulCheckAttempts() % 10 == 0) && checkIgnoreRecord(record)) {
+                  continue;
+               }
                long confirmationTimestamp = record.getCurrentConfirmationTimestamp();
                if (confirmationTimestamp >= 0) {
-                  log.debug("Detected stale read, keyId: " + record.getKeyId());
+                  log.debug("Detected stale read, keyId: " + keyGenerator.generateKey(record.getKeyId()));
                }
                if (confirmationTimestamp >= 0
-                     && (writeApplyMaxDelay <= 0 || System.currentTimeMillis() > confirmationTimestamp + writeApplyMaxDelay)) {
-                  // one more check to see whether some operations should not be ignored
-                  if (ignoreDeadCheckers) {
-                     Object ignored = basicCache.get(ignoredKey(slaveIndex, record.getThreadId()));
-                     if (ignored != null && record.getOperationId() <= (Long) ignored) {
-                        log.debugf("Operations %d - %d for thread %d are ignored.", record.getOperationId(), ignored, record.getThreadId());
-                        while (record.getOperationId() <= (Long) ignored) {
-                           record.next();
-                        }
-                        continue;
-                     }
+                     && (logLogicConfiguration.writeApplyMaxDelay <= 0 || System.currentTimeMillis() > confirmationTimestamp + logLogicConfiguration.writeApplyMaxDelay)) {
+                  // Check one more time to be sure the record should not be ignored
```

Comment:
the comment is a bit misleading - you're not checking 'one more time', you're just retrieving the ignored key